c7n/resources/rds.py

```diff
@@ -356,6 +356,88 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('has-pending-maintenance')
+class HasPendingMaintenance(Filter):
+    """ Scan DB instances for those with pending maintenance
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: rds-pending-maintenance
+                resource: rds
+                filters:
+                  - has-pending-maintenance
+    """
+
+    schema = type_schema('has-pending-maintenance')
+    permissions = ('rds:DescribePendingMaintenanceActions',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+
+        results = []
+        for r in resources:
+            response = client.describe_pending_maintenance_actions(
+                ResourceIdentifier=r['DBInstanceArn'])
```

Comment:
Instead of checking the maintenance for RDS elements one at a time, it might be possible to do them all upfront with something like this:

```
rds.describe_pending_maintenance_actions(Filters=[{'Name': 'db-instance-id', 'Values': [arn1, arn2, ...]}])
```

Reply:
interesting... yeah that seems reasonable as a worthwhile as a tradeoff on api count vs doing per resource api calls.